datafusion/core/src/execution/context.rs

```diff
@@ -1055,12 +1050,16 @@ impl SessionConfig {
         self.set(key, ScalarValue::Boolean(Some(value)))
     }
 
+    /// Set a u32 configuration option
+    pub fn set_u32(self, key: &str, value: u32) -> Self {
+        self.set(key, ScalarValue::UInt32(Some(value)))
+    }
+
     /// Customize batch size
-    pub fn with_batch_size(mut self, n: usize) -> Self {
+    pub fn with_batch_size(self, n: usize) -> Self {
```

Comment:
I like this approach (with a named function `with_batch_size` that calls into the generic implementation below). Seems like a good idea to me

👍